hikari/api/rest.py

```diff
@@ -1210,6 +1210,117 @@ async def create_message(
             If an internal error occurs on Discord while handling the request.
         """
 
+    @abc.abstractmethod
+    async def create_voice_message(
+        self,
+        channel: snowflakes.SnowflakeishOr[channels_.TextableChannel],
+        attachment: files.Resourceish,
+        waveform: str,
+        duration: float,
+        *,
+        reply: undefined.UndefinedOr[snowflakes.SnowflakeishOr[messages_.PartialMessage]] = undefined.UNDEFINED,
+        reply_must_exist: undefined.UndefinedOr[bool] = undefined.UNDEFINED,
+        mentions_reply: undefined.UndefinedOr[bool] = undefined.UNDEFINED,
+        flags: typing.Union[undefined.UndefinedType, int, messages_.MessageFlag] = undefined.UNDEFINED,
+    ) -> messages_.Message:
+        """Create a voice message in the given channel.
+
+        Parameters
+        ----------
+        channel
+            The channel to create the message in.
+        attachment
+            The audio you want to send. This can be a resource,
+            or string of a path on your computer or a URL.
+
+            Its important that it is an audio resource. The content type
+            must start with `audio/`
+
+            Attachments can be passed as many different things, to aid in
+            convenience.
+
+            - If a [`pathlib.PurePath`][] or [`str`][] to a valid URL, the
+                resource at the given URL will be streamed to Discord when
+                sending the message. Subclasses of
+                [`hikari.files.WebResource`][] such as
+                [`hikari.files.URL`][],
+                [`hikari.messages.Attachment`][], etc will also be uploaded this way.
+                This will use bit-inception, so only a small percentage of the
+                resource will remain in memory at any one time, thus aiding in
+                scalability.
+            - If a [`hikari.files.Bytes`][] is passed, or a [`str`][]
+                that contains a valid data URI is passed, then this is uploaded
+                with a randomized file name if not provided.
+            - If a [`hikari.files.File`][], [`pathlib.PurePath`][] or
+                [`str`][] that is an absolute or relative path to a file
+                on your file system is passed, then this resource is uploaded
+                as an attachment using non-blocking code internally and streamed
+                using bit-inception where possible. This depends on the
+                type of [`concurrent.futures.Executor`][] that is being used for
+                the application (default is a thread pool which supports this
+                behaviour).
+        waveform
+            Is intended to be a preview of the entire voice message, with 1 byte
+            per datapoint encoded in base64. Official clients sample the recording
+            at most once per 100 milliseconds, but will downsample so that no more
+            than 256 datapoints are in the waveform.
```

Comment:
```suggestion
            The waveform of the entire message, with 1 byte
            per datapoint encoded in base64.
            
            Official clients sample the recording at most once per 100
            milliseconds, but will downsample so that no more than 256
            datapoints are in the waveform.
            
            !!! note
                Discord states that this is implementation detail and might
                change without notice. You have been warned!
```